pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -124,6 +124,16 @@ func (r *RepoSyncReconciler) Reconcile(ctx context.Context, req controllerruntim
 		return controllerruntime.Result{}, status.APIServerError(err, "failed to get RepoSync")
 	}
 
+	if !rs.DeletionTimestamp.IsZero() {
```

Comment:
I'm wondering whether we should check for the existence of a finalizer here. Was it intentional to only check deletion timestamp?

Reply:
The finalizer doesn't really matter in this case, except that if there's no finalizers then the api server just deletes the object outright, rather than updating it with a deletionTimestamp. So I put the finalizer into the test to better simulate real behavior.

Once https://github.com/GoogleContainerTools/kpt-config-sync/pull/77 is merged, the fake.Client will handle adding the deletionTimestamp when a Delete is requested with a finalizer present, and then also deleting the object when the last finalizer is removed. So I'll have to update the test for that, but didn't want to block on that being merged first.

Reply:
Perhaps more simply: The finalizer causes the deletionTimestamp. So we don't need to check for the finalizer. Without the finalizer, the deletionTimestamp is never added, so that code branch is never executed.

An example is the finalizer that k8s adds when you do a Foreground deletion. Then it looks for owned resource to delete before removing the finalizer. But since our fake.Client doesn't simulate that either yet, I just went with a "test-finalizer" to sort of imply that it doesn't matter what finalizer it is that causes the deletionTimestamp.

Reply:
Makes sense, thank you for explaining!